ui/src/app/cohort-review/set-annotation-item/set-annotation-item.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+
+import {ReviewStateService} from '../review-state.service';
+import {SetAnnotationItemComponent} from './set-annotation-item.component';
+
+import {
+  queryByCss,
+  updateAndTick,
+} from 'testing/test-helpers';
+
+import {
+  AnnotationType,
+  CohortAnnotationDefinition,
+  CohortAnnotationDefinitionService,
+} from 'generated';
+
+class StubRoute {
+  snapshot = {params: {
+    ns: 'workspaceNamespace',
+    wsid: 'workspaceId',
+    cid: 1
+  }};
+}
+
+class ApiSpy {
+  updateCohortAnnotationDefinition = jasmine.createSpy('updateCohortAnnotationDefinition');
+  deleteCohortAnnotationDefinition = jasmine.createSpy('deleteCohortAnnotationDefinition');
+  getCohortAnnotationDefinitions = jasmine.createSpy('getCohortAnnotationDefinitions');
+}
+
+describe('SetAnnotationItemComponent', () => {
+  let fixture: ComponentFixture<SetAnnotationItemComponent>;
+  let component: SetAnnotationItemComponent;
+
+  beforeEach(fakeAsync(() => {
+    TestBed
+      .configureTestingModule({
+        declarations: [
+          SetAnnotationItemComponent,
+        ],
+        imports: [
+          ClarityModule,
+          ReactiveFormsModule,
+        ],
+        providers: [
+          ReviewStateService,
+          {provide: CohortAnnotationDefinitionService, useValue: new ApiSpy()},
+          {provide: ActivatedRoute, useClass: StubRoute},
+        ],
+      }).compileComponents().then((resp) => {
+        fixture = TestBed.createComponent(SetAnnotationItemComponent);
+
+        // tick();
+        component = fixture.componentInstance;
+
+        // Default Inputs for tests
+        component.definition = <CohortAnnotationDefinition>{
+          cohortAnnotationDefinitionId: 1,
+          cohortId: 1,
+          columnName: 'Test Defn',
+          annotationType: AnnotationType.STRING,
+        };
+        updateAndTick(fixture);
+      });
+    // tick();
+  }));
+
+  it('Should render', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('Should set the form focus when editing', fakeAsync(() => {
+    component.editing = true;
+    updateAndTick(fixture);
+    component.setFocus();
+    // component.edit() calls setTimeout so there's still a timer on the stack
+    // without this call to tick
+    tick();
+    const inpElem = component.nameInput.nativeElement;
+    expect(inpElem).toBeTruthy();
+    // Query By css doesn't work here for some reason; just use the real DOM api
+    const focusElem = document.activeElement;
+    expect(focusElem).toBeTruthy();
+    expect(inpElem).toBe(focusElem);
+  }));
+
+  it('Should set the form value to the existing definition value', fakeAsync(() => {
+    // Monkey patch this method b/c it misbehaves in the test environment
+    // unless treated in isolation
+    component.setFocus = () => {};
+    updateAndTick(fixture);
+    component.edit();
+    expect(component.definition.columnName).toEqual(component.name.value);
+  }));
+
+  it('Should refuse to save an empty name value', fakeAsync(() => {
```

Comment:
Should there be a test case where the updateCohortAnnotationDefinition actually gets called?

Reply:
Probably, haha.  I'll push that up next.